src/live_restore/live_restore_server.c

```diff
@@ -0,0 +1,292 @@
+        F_CLR(ctx, WT_THREAD_RUN);
+        __wt_verbose_debug2(session, WT_VERB_FILEOPS, "%s", "Live restore worker terminating");
+        __wt_spin_unlock(session, &server->queue_lock);
+        return (0);
+    }
+
+    WT_LIVE_RESTORE_WORK_ITEM *work_item = NULL;
+
+    work_item = TAILQ_FIRST(&server->work_queue);
+    WT_ASSERT(session, work_item != NULL);
+    TAILQ_REMOVE(&server->work_queue, work_item, q);
+    WT_STAT_CONN_SET(session, live_restore_queue_length, --server->queue_size);
+    __wt_verbose_debug2(
+      session, WT_VERB_FILEOPS, "Live restore worker taking queue item: %s", work_item->uri);
+    __wt_spin_unlock(session, &server->queue_lock);
+
+    WT_CURSOR *cursor;
+    WT_SESSION *wt_session = (WT_SESSION *)session;
+
+    /*
+     * Open a cursor so no one can get exclusive access on the object. This prevents concurrent
+     * schema operations like drop. Even if this object is a log file it can have a cursor opened
+     * it. Opening a cursor on a log will prevent it from getting archived.
+     *
+     * If the file no longer exists, which for logs means they could have been archived and for
+     * regular files dropped, don't error out.
+     */
+    ret = wt_session->open_cursor(wt_session, work_item->uri, NULL, NULL, &cursor);
+    if (ret == ENOENT)
+        return (0);
+    WT_RET(ret);
+
+    /*
+     * We need to get access to the WiredTiger file handle. Given we've opened the cursor we should
+     * be able to access the WT_FH by first getting to its block manager and then the WT_FH.
+     */
+    WT_BM *bm = CUR2BT(cursor)->bm;
+    WT_ASSERT(session, bm->is_multi_handle == false);
+
+    /* FIXME-WT-13897 Replace this with an API call into the block manager. */
+    WT_FILE_HANDLE *fh = bm->block->fh->handle;
+
+    /*
+     * Call the fill holes function. Right now no other reads or writes should be occurring
+     * concurrently or else things will eventually break.
+     *
+     * FIXME-WT-13825: Update this comment.
+     */
+    __wt_verbose_debug2(
+      session, WT_VERB_FILEOPS, "Live restore worker filling holes for: %s", work_item->uri);
+    ret = __wti_live_restore_fs_fill_holes(fh, wt_session);
+    WT_TRET(cursor->close(cursor));
+
+    return (ret);
+}
+
+/*
+ * __live_restore_populate_queue --
+ *     Populate the live restore work queue. Free all objects on failure.
+ */
+static int
+__live_restore_populate_queue(WT_SESSION_IMPL *session, uint64_t *work_count)
+{
+    WT_DECL_RET;
+    WT_LIVE_RESTORE_SERVER *server = &S2C(session)->live_restore_server;
+
+    TAILQ_INIT(&server->work_queue);
+    /*
+     * Open a metadata cursor to gather the list of objects. The metadata file is built from the
+     * WiredTiger.backup file, during __wt_turtle_init. Thus this function must be run after that
+     * function.
+     */
+
+    /*
+     * FIXME-WT-13888: Add logic to queue log files first, then the history store. This will use a
```

Comment:
The worker logs start with "live restore worker". Could we make these start with "live restore server"? To make them easily searchable.

Reply:
e50c7e5dd